### DISCUSSION TITLE
mmcv with cuda 12.1

### DISCUSSION THREAD

**Discussion by isobelm** [open-mmlab/mmcv#2856] [Q&A]
Is it possible to use mmcv with cuda 12.1?  

I am using Ubuntu 22.04.2 I have python 3.8.16, pytorch 2.1.0 and openmim 0.3.9.
 I have been trying to use `mim install "mmcv>=2.0.0"` but it always ends with:

```
  ERROR: Failed building wheel for mmcv
  Running setup.py clean for mmcv
Failed to build mmcv
ERROR: Could not build wheels for mmcv, which is required to install pyproject.toml-based projects
```

**Comment by Vanleya** [comment]
have u sloved yet?

**Reply by samskalicky** [reply]
Hi guys, you can build mmcv with CUDA 12 by following the instructions to build from source. heres a couple pointers:
https://github.com/open-mmlab/mmcv/?tab=readme-ov-file#install-mmcv
https://mmcv.readthedocs.io/en/latest/get_started/build.html
https://github.com/open-mmlab/mmcv/blob/761d780a24c118f383868d4d78fd3fe429ae95d5/docker/dev/Dockerfile#L28-L31
https://github.com/open-mmlab/mmcv/blob/761d780a24c118f383868d4d78fd3fe429ae95d5/setup.py#L159-L170

heres some reference steps to customize the CUDA build (ie. to build for the latest Nvidia H200 GPU with CUDA 9.0 arch)
```
git clone https://github.com/open-mmlab/mmcv.git
cd mmcv
export FORCE_CUDA=1
export MMCV_CUDA_ARGS="-gencode arch=compute_90,code=compute_90
pip install -e .
```
